File: Java/ArrayListAndStreamPractice/src/Anagrams/AnagramValidationTest.java
package Anagrams;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AnagramValidationTest {

    AnagramValidation test1;

    AnagramValidation test2;

    AnagramValidation test3;

    @BeforeEach
    void setUp() {
        this.test1 = new AnagramValidation("listen", "silent");
        this.test2 = new AnagramValidation("astronomer", "moonstarer");
        this.test3 = new AnagramValidation("hahaha", "blahblahschmah");
    }

    @Test
    void validateByMeansOfStringComparison() {
        assertTrue(this.test1.validateByMeansOfStringComparison());
        assertTrue(this.test2.validateByMeansOfStringComparison());
        assertFalse(this.test3.validateByMeansOfStringComparison());
    }

    @Test
    void validateByMeansOfMapComparison() {
        assertTrue(this.test1.validateByMeansOfMapComparison());
        assertTrue(this.test2.validateByMeansOfMapComparison());
        assertFalse(this.test3.validateByMeansOfMapComparison());
    }
}

File: Kotlin/higherOrderArrayMethods/test/anagrams/AnagramValidationTest.kt
package anagrams

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class AnagramValidationTest {

    private var validation1 = AnagramValidation("app", "ppa")
    private var validation2 = AnagramValidation("astronomer", "moonstarer")
    private var validation3 = AnagramValidation("silent", "listen")
    private var validation4 = AnagramValidation("whohjos", "ljdljadsas")

    @Test
    fun validateByMeansOfMapComparison() {
        assertTrue(validation1.validateByMeansOfMapComparison())
        assertTrue(validation2.validateByMeansOfMapComparison())
        assertTrue(validation3.validateByMeansOfMapComparison())
        assertFalse(validation4.validateByMeansOfMapComparison())
    }

    @Test
    fun validateByMeansOfStringComparison() {
        assertTrue(validation1.validateByMeansOfStringComparison())
        assertTrue(validation2.validateByMeansOfStringComparison())
        assertTrue(validation3.validateByMeansOfStringComparison())
        assertFalse(validation4.validateByMeansOfStringComparison())
    }
}